src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -259,64 +250,50 @@ public static void anchorBlocks(EntityPlayer player, ItemStack stack) {
-            String[] directionNames = NBTKeys.GADGET_VALUES;
-            AxisAlignedBB area = new AxisAlignedBB(pos);
-            for (int i = 0; i < directionNames.length; i++)
-                area = area.union(new AxisAlignedBB(pos.offset(directions.get(i), getToolValue(stack, directionNames[i]) - (i == 4 ? 1 : 0))));
+            Region boundary = getClearingRegion(pos, incomingSide, player, stack);
+            HashSet<BlockPos> voidPositions = new HashSet<>();
+            int depth = getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH);
+            if (depth == 0)
+                return new SetBackedPlacementSequence(voidPositions, boundary);
+
+            BlockPos startPos = (getAnchor(stack) == null) ? pos : getAnchor(stack);
+            IBlockState stateTarget = !Config.GADGETS.GADGET_DESTRUCTION.nonFuzzyEnabled.get() || GadgetGeneric.getFuzzy(stack) ? null : world.getBlockState(pos);
+            addConnectedCoordinates(world, player, startPos, stateTarget, voidPositions,
+                    boundary);
 
-            addConnectedCoords(world, player, startPos, stateTarget, voidPositions,
-                    (int) area.minX, (int) area.minY, (int) area.minZ, (int) area.maxX - 1, (int) area.maxY - 1, (int) area.maxZ - 1);
+            return new SetBackedPlacementSequence(voidPositions, boundary);
         } else {
-            int left = -getToolValue(stack, NBTKeys.GADGET_VALUE_LEFT);
-            int right = getToolValue(stack, NBTKeys.GADGET_VALUE_RIGHT);
-            int down = -getToolValue(stack, NBTKeys.GADGET_VALUE_DOWN);
-            int up = getToolValue(stack, NBTKeys.GADGET_VALUE_UP);
-            for (int d = 0; d < depth; d++) {
-                for (int x = left; x <= right; x++) {
-                    for (int y = down; y <= up; y++) {
-                        BlockPos voidPos = new BlockPos(startPos);
-                        voidPos = voidPos.offset(directions.get(0), x);
-                        voidPos = voidPos.offset(directions.get(2), y);
-                        voidPos = voidPos.offset(directions.get(4), d);
-                        if (validBlock(world, voidPos, player, stateTarget))
-                            voidPositions.add(voidPos);
-                    }
-                }
-            }
+            return getClearingRegion(pos, incomingSide, player, stack);
         }
-        return voidPositions;
     }
 
-    public static void addConnectedCoords(World world, EntityPlayer player, BlockPos loc, IBlockState state,
-            SortedSet<BlockPos> coords, int minX, int minY, int minZ, int maxX, int maxY, int maxZ) {
-        if (coords.contains(loc) || loc.getX() < minX || loc.getY() < minY || loc.getZ() < minZ || loc.getX() > maxX || loc.getY() > maxY || loc.getZ() > maxZ)
-            return;
+    public static SortedSet<BlockPos> getClearingPositionsSet(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
```

Comment:
Is this used anywhere? I can't seem to find a usage of this Method.

Reply:
It is used in `ToolRenders`, in where it called `SortedSet<BlocPos> getArea(World, BlockPos, EnumFacing, EntityPlayer, ItemStack)` originally.

Reply:
It may be possible to refactor ToolRenders to use an Iterable, so that we can avoid the overhead of iterating over this possibly large PlacementSequence.

Reply:
We have to sort the set anyway when rendering, so Iterable don't really help.

Reply:
I'm still wondering why rendering needs these to be sorted... Well, so be it...